shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,392 @@
+      return true;
+    }
+  }
+
+  private static class SshValidationTask implements ValidationTask {
+    private final String mFileName;
+
+    public SshValidationTask(String fileName) {
+      mFileName = fileName;
+    }
+
+    @Override
+    public boolean validate() {
+      List<String> nodes = readNodeList(mFileName);
+      if (nodes == null) {
+        return false;
+      }
+
+      boolean hasUnreachableNodes = false;
+      for (String nodeName : nodes) {
+        if (!isAddressReachable(nodeName, 22)) {
+          System.err.format("Unable to reach ssh port 22 on node %s.%n", nodeName);
+          hasUnreachableNodes = true;
+        }
+      }
+      return !hasUnreachableNodes;
+    }
+  }
+
+  // read a list of nodes from given file name ignoring comments and empty lines
+  private static List<String> readNodeList(String fileName) {
+    String confDir = Configuration.get(PropertyKey.CONF_DIR);
+    List<String> lines;
+    try {
+      lines = Files.readAllLines(Paths.get(confDir, fileName), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      System.err.format("Unable to read file %s/%s.%n", confDir, fileName);
+      return null;
+    }
+
+    List<String> nodes = new ArrayList<>();
+    for (String line : lines) {
+      String node = line.trim();
+      if (node.startsWith("#") || node.length() == 0) {
+        continue;
+      }
+      nodes.add(node);
+    }
+
+    return nodes;
+  }
+
+  private static boolean validateRemote(List<String> nodes, String target, String name)
+      throws InterruptedException {
+    if (nodes == null) {
+      return false;
+    }
+
+    boolean success = true;
+    for (String node : nodes) {
+      success &= validateRemote(node, target, name);
+    }
+
+    return success;
+  }
+
+  // validate environment on remote node
+  private static boolean validateRemote(String node, String target, String name)
+      throws InterruptedException {
+    System.out.format("Validating %s environment on %s...%n", target, node);
+    if (isAddressReachable(node, 22)) {
+      String homeDir = Configuration.get(PropertyKey.HOME);
+      String remoteCommand = String.format(
+              "%s/bin/alluxio validateEnv %s %s", homeDir, target, name == null ? "" : name);
+      String localCommand = String.format(
+              "ssh -o ConnectTimeout=5 -o StrictHostKeyChecking=no -tt %s \"bash %s\"",
+              node, remoteCommand);
+      String[] command = {"bash", "-c", localCommand};
+      try {
+        ProcessBuilder builder =
+                new ProcessBuilder(command);
+        builder.redirectErrorStream(true);
+        builder.redirectOutput(ProcessBuilder.Redirect.INHERIT);
+        builder.redirectInput(ProcessBuilder.Redirect.INHERIT);
+        Process process = builder.start();
+        process.waitFor();
+        return process.exitValue() == 0;
+      } catch (IOException e) {
+        System.err.format("Unable to validate on node %s: %s.%n", node, e.getMessage());
```

Comment:
I set up `ProcessBuilder` to redirect both input and output so that stdout and stderr will show up on screen or whatever output stream `validateEnv` command is set up with.